### PR TITLE
Server-side UI Defaults

### PR DESCRIPTION
**Pull request by cfcosta**
Part of https://github.com/kiali/kiali/issues/3371.

Please see: https://github.com/kiali/kiali-ui/pull/2064

### CODE REVIEW

File: roles/default/kiali-deploy/defaults/main.yml [hunk 1]
```diff
@@ -187,6 +187,10 @@ kiali_defaults:
 
   kiali_feature_flags:
     istio_injection_action: true
+    ui_defaults:
+      namespaces: []
+      metrics_per_refresh: "1m"
+      auto_refresh_interval: "15s"
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:193]: Only request I have is to make sure the peer settings are alphabetically ordered:
```suggestion
    ui_defaults:
      auto_refresh_interval: "15s"
      metrics_per_refresh: "1m"
      namespaces: []
```

File: deploy/kiali/kiali_cr.yaml [hunk 2]
```diff
@@ -642,6 +642,26 @@ spec:
...
+# Number of metrics to get on each update. Omit for default.
+# Valid values: 1m, 5m, 10m, 30m, 1h, 3h, 6h, 12h, 1d, 7d, 30d
+#        ---
+#        metrics_per_refresh: "1m"
+#
+# How long should the interval be between each graph refresh. Omit for default,
+# "pause" to disable auto refresh.
+# Valid values: pause, 10s, 15s, 30s, 1m, 5m, 15m
+#        ---
+#        auto_refresh_interval: "15s"
```
Comment (jmazzitelli) [inline deploy/kiali/kiali_cr.yaml:664]: Only request I have is to make sure the peer settings are alphabetically ordered:
```suggestion
# How long should the interval be between each graph refresh. Omit for default,
# "pause" to disable auto refresh.
# Valid values: pause, 10s, 15s, 30s, 1m, 5m, 15m
#        ---
#        auto_refresh_interval: "15s"
#
# Number of metrics to get on each update. Omit for default.
# Valid values: 1m, 5m, 10m, 30m, 1h, 3h, 6h, 12h, 1d, 7d, 30d
#        ---
#        metrics_per_refresh: "1m"
#
# Defines the default namespaces selected when the graph page opens, set it to
# empty array (or omit key) for no default.
#        ---
#        namespaces: []
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Review by jmazzitelli — DISMISSED** [review]
LGTM

File: deploy/kiali/kiali_cr.yaml [hunk 3]
```diff
@@ -642,6 +642,26 @@ spec:
...
+#
+# Defines defaults for the UI
+#    ---
+#    ui_defaults:
+#
+# How long should the interval be between each graph refresh. Omit for default,
+# "pause" to disable auto refresh.
+# Valid values: pause, 10s, 15s, 30s, 1m, 5m, 15m
+#        ---
+#        auto_refresh_interval: "15s"
```
Comment (jshaughn) [inline deploy/kiali/kiali_cr.yaml:654]: This setting applies to more than just the graph.  It applies, I think, to every page offering auto refresh.  Maybe just change to:

` The automatic refresh interval for pages offering automatic refresh.`

Also, I think "automatic" is implied, so you could opt to remove "auto_".

File: deploy/kiali/kiali_cr.yaml [hunk 4]
```diff
@@ -642,6 +642,26 @@ spec:
...
+#
+# Number of metrics to get on each update. Omit for default.
+# Valid values: 1m, 5m, 10m, 30m, 1h, 3h, 6h, 12h, 1d, 7d, 30d
+#        ---
+#        metrics_per_refresh: "1m"
```
Comment (jshaughn) [inline deploy/kiali/kiali_cr.yaml:659]: I'd change 'Number of...' to `Duration of..` or `Time period of...`  `..metrics to fetch on each refresh`.

File: deploy/kiali/kiali_cr.yaml [hunk 5]
```diff
@@ -642,6 +642,26 @@ spec:
...
+#
+# Defines the default namespaces selected when the graph page opens, set it to
+# empty array (or omit key) for no default.
+#        ---
+#        namespaces: []
```
Comment (jshaughn) [inline deploy/kiali/kiali_cr.yaml:664]: Again, this has greater scope than the graph.  Something like:

`Default selections for the Namespace Selection dropdown.  Non-existent of inaccessible namespaces will be ignored. 
 Omit, or set to an empty array, for no default namespaces.`

**Review by jshaughn — CHANGES_REQUESTED** [review]
Good intial settings.  A few minor changes needed.

**Comment by cfcosta** [comment]
@jshaughn added your feedback, will rename the keys to remove the `auto_` from the name, as it is indeed implied.

File: deploy/kiali/kiali_cr.yaml [hunk 6]
```diff
@@ -642,6 +642,28 @@ spec:
 # Flag to indicate Kiali to enable/disable an Action to label a namespace for automatic Istio Sidecar injection.
 #    ---
 #    istio_injection_action: true
+#
+# Default settings for the UI. These defaults apply to all users.:654
+#
```
Comment (jmazzitelli) [inline deploy/kiali/kiali_cr.yaml:647]: Remove odd characters. No need for blank line before the "---" line.
```suggestion
# Default settings for the UI. These defaults apply to all users.
```

File: roles/default/kiali-deploy/defaults/main.yml [hunk 7]
```diff
@@ -14,18 +14,18 @@ kiali_defaults:
   version: "default"
 
   additional_display_details:
-  - title: "API Documentation"
-    annotation: "kiali.io/api-spec"
-    icon_annotation: "kiali.io/api-type"
+    - title: "API Documentation"
+      annotation: "kiali.io/api-spec"
+      icon_annotation: "kiali.io/api-type"
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:19]: Revert this file - the indentation is screwed up.

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

File: deploy/kiali/kiali_cr.yaml [hunk 8]
```diff
@@ -653,6 +653,27 @@ spec:
...
+# Default selections for the namespace selection dropdown. Non-existent or
+# inaccessible namespaces will be ignored. Omit, or set to an empty array for no
+# default namespaces.
+#      ---
+#      namespaces: []
+#
+# Duration of metrics to fetch on each refresh. Omit for default.
+# Valid values: 1m, 5m, 10m, 30m, 1h, 3h, 6h, 12h, 1d, 7d, 30d
+#      ---
+#      metrics_per_refresh: "1m"
```
Comment (jmazzitelli) [inline deploy/kiali/kiali_cr.yaml:670]:
```suggestion
# Duration of metrics to fetch on each refresh. Omit for default.
# Valid values: 1m, 5m, 10m, 30m, 1h, 3h, 6h, 12h, 1d, 7d, 30d
#      ---
#      metrics_per_refresh: "1m"
#241 # Default selections for the namespace selection dropdown. Non-existent or
# inaccessible namespaces will be ignored. Omit, or set to an empty array for no
# default namespaces.
#      ---
#      namespaces: []
```

File: roles/default/kiali-deploy/defaults/main.yml [hunk 9]
```diff
@@ -189,6 +189,10 @@ kiali_defaults:
 
   kiali_feature_flags:
     istio_injection_action: true
+    ui_defaults:
+      namespaces: []
+      metrics_per_refresh: "1m"
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:194]:
```suggestion
      metrics_per_refresh: "1m"
      namespaces: []
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
opps... sort the settings

**Review by jmazzitelli — APPROVED** [review]

**Review by jmazzitelli — APPROVED** [review]
I ran tests - this works.

**Review by jshaughn — APPROVED** [review]
LG

**Review by pbajjuri20 — APPROVED** [review]

**Pull request merged**